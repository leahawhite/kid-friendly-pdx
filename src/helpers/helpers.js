/*export function numberOfReviews(place) {
  const placeReviews = reviews.filter(review.place_id == place.id)
  return placereviews.length
}

export function averageReviewRating(place) {
  const placeReviews = reviews.filter(review.place_id == place.id)
  const ratingArray = placeReviews.map(review => review.rating) 
  const ratingsSum = ratingArray.reduce(function(a,b){return a + b})
  const numReviews = this.numberOfReviews(place)
  return ratingsSum / numReviews
}*/

export function readableReviewCount(number) {
  switch(number) {
    case 0:
      return 'no reviews yet'

    case 1:
      return `1 review`

    default:
      return `${number} reviews`
  }
}

export const testPlace = {
  id: 1,
  name: "Atlas Pizza",
  address: "6529 SE Foster Rd",
  city: "Portland",
  state: "OR",
  zipcode: "97206",
  lat: 45.489200,
  lng: -122.595070,
  neighborhood: "SE",
  phone: "503-232-3004",
  website: "http://atlaspizzapdx.com",
  hours: [
    {
      "dayOfWeek": "Monday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Tuesday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Wednesday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Thursday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Friday",
      "opens": "12:00 pm",
      "closes": "12:00 am",
    },
    {
      "dayOfWeek": "Saturday",
      "opens": "12:00 pm",
      "closes": "12:00 am",
    },
    {
      "dayOfWeek": "Sunday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
  ],
  date_created: "2019-06-13",
  category: ["restaurants",],
  descriptors: [ "pizza", "arcade", "beer" ],
  features: { 
    "toys": true, 
    "play area": true, 
    "arcade": true,
    "quick service": true,
    "all ages": true,
    "comfortable seating": true,
    "friendly staff": true,
    "kids menu": false,
    "highchairs/boosters": false,
    "changing station": false,
    "flexible": true,
    "patio/sidewalk": true,
  },
  images: [
    {
      id: 1,
      src: 'https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400',
      title: `Atlas Pizza's 'Jalapeno Popper' pie`,
      place_id: 1,
      user_id: 2,
      review_id: 1,
    },
    {
      id: 2,
      src: 'https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400',
      title: 'A rainbow over Foster Blvd',
      place_id: 1,
      user_id: 1,
      review_id: 2,
    },
  ],
  number_of_reviews: 2,
  average_review_rating: 4.5,
};

export const testReviews = [
  { 
    id: 1,
    star_rating: 5,
    text: "Atlas is our new favorite spot for a quick, painless dinner with our toddler. It's super casual -- order slices or pies at the counter, and they call your name for pickup when it's ready. Since it's new, the play area is pretty clean and well-stocked. There's an old church pew there for parents to sit and supervise, but the open floor plan means that you can see the play area from pretty much every table. We've only ordered slices, so we've never had to wait long. We're usually in and out in 30 minutes, which is just about perfect for us.",
    date_created: "2019-06-13",
    place_id: 1,
    user_id: 5,
    images: [
      {
        id: 1,
        src: 'https://lh3.googleusercontent.com/QDwk6G6ZCwYzVH-yJHQ824GZ1wKW2XXus8hxxdH-4AAHz-FejC-UVydyE5le1ZvPTIo1cXw1G25qhljwS8dgyIedCCFtIxBonznVQ4LfaPDEZVWmqKS1vFqVUFFNmMmMlKQ5eCpfhNI=w2400',
        title: `Atlas Pizza's play area`,
        place_id: 1,
        user_id: 5,
      },
    ]
  },
  { 
    id: 2,
    star_rating: 4,
    text: "Very easy place to take kids of all ages. Nice toy area with a great variety of toys!",
    date_created: "2019-06-13",
    place_id: 1,
    user_id: 1,
    images: [],
  },
];



