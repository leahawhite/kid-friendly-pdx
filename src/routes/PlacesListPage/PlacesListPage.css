.places-list {
  display: flex;
  flex-direction: column;
  align-content: center;
  padding: 0 10px;
  min-height: 80vh;
  position: relative;
}

.places-header {
  display: flex;
  justify-content: space-between;
  align-content: center;
  padding-bottom: 20px;
}

.sort-results-form {
  margin: 0;
  display: flex;
  flex-direction: column;
  text-align: left;
}

.sort-results {
  margin-right: 10px;
  font-size: 18px;
}

.sort-select {
  padding: 5px 8px;
  border: 3px solid #00b8e6;
  background-color: white;
  border-radius: 8px;
  font-size: 16px;
  font-family: 'Source Sans Pro';
}

/* .no-results {
  padding: 10px 0;
} */

.places-map {
  height: 250px;
  margin-bottom: 20px;
  width: auto;
}

@media only screen and (min-width: 950px) {
  .places-list {
    width: 70%;
    margin: 0 auto;
  }

  .sort-results-form {
    margin: 0;
    display: block;
  }
}
