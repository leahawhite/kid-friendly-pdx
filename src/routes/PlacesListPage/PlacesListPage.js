import React, { Component } from 'react';
import SearchBar from '../../components/SearchBar/SearchBar';
import PlacesListItem from '../../components/PlacesListItem/PlacesListItem';
import Map from '../../components/MapContainer/Map';
import data from '../../data';
import './PlacesListPage.css';

export default class PlacesListPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      places: [],
    }
  }

  componentDidMount() {
    // fetch will happen here
    this.setState(data);
  }

  renderPlaces() {
    const { places } = this.state
    const { searchTerm, category, neighborhood } = this.props.location.state
    console.log('places', places)
    console.log('searchTerm', searchTerm)
    const results = places.filter(place => (place.name.toLowerCase().includes(searchTerm.toLowerCase())
      || place.descriptors.includes(searchTerm.toLowerCase())) 
      && (category === place.category || category === 'all') && (neighborhood === place.neighborhood || neighborhood === 'All Portland'))
    console.log('results', results)
    
    return (
      results.length ?
        results.map(place => 
        <PlacesListItem key={place.id} place={place} /> ) 
        : <p>Sorry, your search returned no results. Please try again.</p>
    )
  }

  renderMap() {
    // repeats the same stuff as renderPlaces -- combine? But map is result array and place is result list.
    const { places } = this.state
    const { searchTerm, category, neighborhood } = this.props.location.state
    const results = places.filter(place => (place.name.toLowerCase().includes(searchTerm.toLowerCase())
      || place.descriptors.includes(searchTerm.toLowerCase())) 
      && (category === place.category || category === 'all') && (neighborhood === place.neighborhood || neighborhood === 'All Portland'))
    console.log('map', results)  
    return (
      results.length ?
      <section className="places-map">
        <Map places={results}/>
      </section> 
        : null
    )
  }
  
  /*handleUpdateSort(sort) {
      this.setState({
        sortResults: sort,
      })
  }*/
  
  render() {
   
    return (
      <>
        <SearchBar />
        <section className="places-list">
          <header className="places-header">
            <h2>Results</h2>
            <form className="sort-results-form">
              <label className="sort-results" htmlFor="sort-results">Sort by:</label>
              <select 
                className="sort-select" 
                type="text" 
                id="sort-results" 
                /*value=""
                /* onChange={e => this.handleUpdateSort(e.target.value)}*/>
                  <option value="distance">distance</option>
                  <option value="rating">rating</option>
              </select>
            </form>
          </header>
          {this.renderPlaces()}
          {this.renderMap()}
        </section>
      </>
    )
  }
}