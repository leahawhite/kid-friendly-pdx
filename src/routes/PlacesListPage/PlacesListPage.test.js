import React from 'react'
import ReactDOM from 'react-dom'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { testPlace } from '../../helpers/helpers'
import PlacesListPage from './PlacesListPage'

describe('PlacesListPage component', () => {
  const searchTerm = 'pizza'
  const category = ''
  const neighborhood = ''
  const places = [testPlace]
  const handleSubmit = jest.fn()
  const handleUpdateCategory = jest.fn()
  const handleUpdateNeighborhood = jest.fn()
  const handleUpdateSearch = jest.fn()
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <PlacesListPage
          searchTerm={searchTerm}
          category={category}
          neighborhood={neighborhood}
          handleUpdateSearch={handleUpdateSearch}
          handleUpdateCategory={handleUpdateCategory}
          handleUpdateNeighborhood={handleUpdateNeighborhood}
          handleSubmit={handleSubmit}
          places={places} />
      </MemoryRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<PlacesListPage
      searchTerm={searchTerm}
      category={category}
      neighborhood={neighborhood}
      handleUpdateSearch={handleUpdateSearch}
      handleUpdateCategory={handleUpdateCategory}
      handleUpdateNeighborhood={handleUpdateNeighborhood}
      handleSubmit={handleSubmit}
      places={places}
    />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
