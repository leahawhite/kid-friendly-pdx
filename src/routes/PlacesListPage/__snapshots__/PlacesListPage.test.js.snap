// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesListPage component renders UI as expected 1`] = `
<Fragment>
  <SearchBar
    category=""
    handleSubmit={[MockFunction]}
    handleUpdateCategory={[MockFunction]}
    handleUpdateNeighborhood={[MockFunction]}
    handleUpdateSearch={[MockFunction]}
    neighborhood=""
    searchTerm="pizza"
  />
  <section
    className="places-list"
  >
    <header
      className="places-header"
    >
      <h2>
        Results
      </h2>
      <form
        className="sort-results-form"
      >
        <label
          className="sort-results"
          htmlFor="sort-results"
        >
          Sort by:
        </label>
        <select
          aria-label="sort results"
          className="sort-select"
          id="sort-results"
          name="sort"
          onChange={[Function]}
          type="text"
        >
          <option
            aria-label="rating"
            name="sort"
            value="rating"
          >
            rating
          </option>
          <option
            aria-label="number of reviews"
            name="sort"
            value="number of reviews"
          >
            number of reviews
          </option>
          <option
            aria-label="distance"
            name="sort"
            value="distance"
          >
            distance
          </option>
        </select>
      </form>
    </header>
    <div
      className="error"
    />
    <PlacesListItem
      key="1"
      place={
        Object {
          "address": "6529 SE Foster Rd",
          "average_review_rating": 4.5,
          "category": Array [
            "restaurants",
          ],
          "city": "Portland",
          "date_created": "2019-06-13",
          "descriptors": Array [
            "pizza",
            "arcade",
            "beer",
          ],
          "features": Object {
            "all ages": true,
            "arcade": true,
            "changing station": false,
            "comfortable seating": true,
            "flexible": true,
            "friendly staff": true,
            "highchairs/boosters": false,
            "kids menu": false,
            "patio/sidewalk": true,
            "play area": true,
            "quick service": true,
            "toys": true,
          },
          "hours": Array [
            Object {
              "closes": "11:00 pm",
              "dayOfWeek": "Monday",
              "opens": "12:00 pm",
            },
            Object {
              "closes": "11:00 pm",
              "dayOfWeek": "Tuesday",
              "opens": "12:00 pm",
            },
            Object {
              "closes": "11:00 pm",
              "dayOfWeek": "Wednesday",
              "opens": "12:00 pm",
            },
            Object {
              "closes": "11:00 pm",
              "dayOfWeek": "Thursday",
              "opens": "12:00 pm",
            },
            Object {
              "closes": "12:00 am",
              "dayOfWeek": "Friday",
              "opens": "12:00 pm",
            },
            Object {
              "closes": "12:00 am",
              "dayOfWeek": "Saturday",
              "opens": "12:00 pm",
            },
            Object {
              "closes": "11:00 pm",
              "dayOfWeek": "Sunday",
              "opens": "12:00 pm",
            },
          ],
          "id": 1,
          "images": Array [
            Object {
              "id": 1,
              "place_id": 1,
              "review_id": 1,
              "src": "https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400",
              "title": "Atlas Pizza's 'Jalapeno Popper' pie",
              "user_id": 2,
            },
            Object {
              "id": 2,
              "place_id": 1,
              "review_id": 2,
              "src": "https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400",
              "title": "A rainbow over Foster Blvd",
              "user_id": 1,
            },
          ],
          "lat": 45.4892,
          "lng": -122.59507,
          "name": "Atlas Pizza",
          "neighborhood": "SE",
          "number_of_reviews": 2,
          "phone": "503-232-3004",
          "state": "OR",
          "website": "http://atlaspizzapdx.com",
          "zipcode": "97206",
        }
      }
    />
    <section
      className="places-map"
    >
      <Map
        center={
          Object {
            "lat": 45.5155,
            "lng": -122.6793,
          }
        }
        infoClass="infowindow"
        places={
          Array [
            Object {
              "address": "6529 SE Foster Rd",
              "average_review_rating": 4.5,
              "category": Array [
                "restaurants",
              ],
              "city": "Portland",
              "date_created": "2019-06-13",
              "descriptors": Array [
                "pizza",
                "arcade",
                "beer",
              ],
              "features": Object {
                "all ages": true,
                "arcade": true,
                "changing station": false,
                "comfortable seating": true,
                "flexible": true,
                "friendly staff": true,
                "highchairs/boosters": false,
                "kids menu": false,
                "patio/sidewalk": true,
                "play area": true,
                "quick service": true,
                "toys": true,
              },
              "hours": Array [
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Monday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Tuesday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Wednesday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Thursday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "12:00 am",
                  "dayOfWeek": "Friday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "12:00 am",
                  "dayOfWeek": "Saturday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Sunday",
                  "opens": "12:00 pm",
                },
              ],
              "id": 1,
              "images": Array [
                Object {
                  "id": 1,
                  "place_id": 1,
                  "review_id": 1,
                  "src": "https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400",
                  "title": "Atlas Pizza's 'Jalapeno Popper' pie",
                  "user_id": 2,
                },
                Object {
                  "id": 2,
                  "place_id": 1,
                  "review_id": 2,
                  "src": "https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400",
                  "title": "A rainbow over Foster Blvd",
                  "user_id": 1,
                },
              ],
              "lat": 45.4892,
              "lng": -122.59507,
              "name": "Atlas Pizza",
              "neighborhood": "SE",
              "number_of_reviews": 2,
              "phone": "503-232-3004",
              "state": "OR",
              "website": "http://atlaspizzapdx.com",
              "zipcode": "97206",
            },
          ]
        }
        zoom={11}
      />
    </section>
  </section>
</Fragment>
`;
